file-share-service-desktop-client/Modules/Admin/JobViewModels/NewBatchJobViewModel.cs

```diff
@@ -269,29 +296,42 @@ protected internal override async Task OnExecuteCommand()
                                         });
                                         if (fileUploadProgressViewModel.CompleteBlocks == fileUploadProgressViewModel.TotalBlocks)
                                         {
-                                            logger.LogInformation("File Share Service upload files completed for file:{file} and BatchId:{BatchId} .", f.file.Name ,batchHandle.BatchId);
+                                            logger.LogInformation("File Share Service upload files completed for file:{file} and BatchId:{BatchId} .", f.file.Name, batchHandle.BatchId);
                                         }
-                                    }).ContinueWith(_ => openRead.Dispose());                            
+                                    }, cancellationToken).ContinueWith(_ => openRead.Dispose());
                             }).ToArray());
                     //cleaning up file progress as all uploaded
                     FileUploadProgress.Clear();
+                    cancellationToken.ThrowIfCancellationRequested();
 
                     logger.LogInformation("File Share Service batch commit started for batch ID:{BatchId}.",batchHandle.BatchId);
                     ExecutionResult = $"Files uploaded, batch commit in progress. New batch ID: {batchHandle.BatchId}";
-                    IsCommitting = true;                
+                    IsCommitting = !IsCanceled;
+                    IsCommittingOnCancel = true;
+                    
                     await fileShareClient.CommitBatch(batchHandle);
```

Comment:
I would prefer that a CancellationToken parameter was added to `fileShareClient.CommitBatch()` but can see that it's unlikely the user will click [Cancel] in the time period between `cancellationToken.ThrowIfCancellationRequested();` and `fileShareClient.CommitBatch()`.

Reply:
Thanks Usha for your explanation. No further action/changes needed.